controllers/operands/deploymentHandler_test.go

```diff
@@ -0,0 +1,106 @@
+package operands
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/common"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/commontestutils"
+)
+
+var _ = Describe("Deployment Handler", func() {
+	Context("update or recreate the Deployment as required", func() {
+		var hco *hcov1beta1.HyperConverged
+		var req *common.HcoRequest
+		var originalDeployment *appsv1.Deployment
+		var modifiedDeployment *appsv1.Deployment
+
+		BeforeEach(func() {
+			hco = commontestutils.NewHco()
+			req = commontestutils.NewReq(hco)
+			originalDeployment = &appsv1.Deployment{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Deployment",
+					APIVersion: "apps/v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "modifiedDeployment",
+					Labels: map[string]string{"key1": "value1"},
+				},
+				Spec: appsv1.DeploymentSpec{},
+			}
+			modifiedDeployment = &appsv1.Deployment{}
+			originalDeployment.DeepCopyInto(modifiedDeployment)
+		})
+
+		It("should recreate the Deployment as LabelSelector has changed", func() {
+			// create a fake client using original deployment
+			cl := commontestutils.InitClient([]runtime.Object{originalDeployment})
```

Comment:
[]client.Object{…} from "sigs.k8s.io/controller-runtime/pkg/client"

Reply:
Fixed. I am running `make sanity` as well as `golangci-lint` locally before pushing. Yet this showed up only in CI, not locally. Took me a bit to realize that I might need to rebase to see the failures locally. :)